include/deal.II/grid/reference_cell.h

```diff
@@ -91,6 +91,33 @@ namespace ReferenceCell
       }
   }
 
+  /**
+   * Retrieve the correct ReferenceCell::Type for a given structural dimension
+   * and number of vertices.
+   */
+  inline Type
+  n_vertices_to_type(const int dim, const unsigned int n_vertices)
+  {
+    AssertIndexRange(dim, 4);
+    AssertIndexRange(n_vertices, 9);
+    const auto X = Type::Invalid;
+
+    static constexpr std::array<std::array<ReferenceCell::Type, 9>, 4> table = {
+      {// dim 0
+       {{X, Type::Vertex, X, X, X, X, X, X, X}},
```

Comment:
Any reason the first value is always invalid?

Reply:
The array indices are the number of vertices in a geometry object. Since no element has zero vertices the first entry is always `X`.